### ISSUE TITLE
fish crashes if there is a file that starts with ^? (delete)

### ISSUE THREAD

**Issue by psbanka** [fish-shell/fish-shell#4515]
fish version: 2.6.0
OS: Darwin 17.0.0 Darwin Kernel Version 17.0.0; root:xnu-4570.1.46~2/RELEASE_X86_64 x86_64
$TERM: xterm-256color

Steps to reproduce:
1. in bash: `touch ^?n` <-- note that the `^?` is created by typing `C-v` `backspace`
2. in the same directory, using fish, perform `ls <tab>`

Fish crashes with: 
```
ls libc++abi.dylib: terminating with uncaught exception of type std::length_error: basic_string
```

**Comment by zanchey** [comment]
You can reproduce this without using bash by creating a file called `\x7fn` - ^? is delete rather than backspace.

I can reproduce this in 2.6.0 but not in 2.7b1, so I think it's been fixed - any chance you could try that version? If you're using Homebrew, I've just submitted the patch so that `brew install --devel fish` will give you 2.7b1 - it might be worth waiting for that to become available.

**Comment by zanchey** [comment]
`brew install --devel fish` now installs 2.7b1, so if you have a chance to test that would be great.

**Comment by faho** [comment]
I would expect this to be a duplicate of #4179, which is why it's not happening with 2.7.0 anymore.

**Comment by psbanka** [comment]
Confirmed fixed in `2.7b1`. Thanks, and sorry for the dup! 😄
